### PR TITLE
.Net: New CreateFromType and CreateMetadataFromType methods

### LINKED ISSUE #8946: .Net: New CreateFromType methods
Processes need to be able to create a `KernelPlugin` from a `Type` and also create the `KernelFunctionMetadata` without creating an instance of the class because at the time the metadata is required not all dependencies will be available.

- [ ] CreateFromType that takes a Type parameter rather than the generic
- [ ] Create collection of KernelFunctionMetadata from a Type parameter or a generic

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Motivation and Context

Closes #8946 

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/SemanticKernel.Core/Functions/KernelFunctionMetadataFactory.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+/// <summary>
+/// Provides factory methods for creating collections of <see cref="KernelFunctionMetadata"/>, such as
+/// those backed by a prompt to be submitted to an LLM or those backed by a .NET method.
+/// </summary>
+[Experimental("SKEXP0001")]
+public static class KernelFunctionMetadataFactory
+{
+    /// <summary>
+    /// Creates a plugin that wraps and instance the specified type that is created using the specified delegate.
```
Comment (alliscode) [inline dotnet/src/SemanticKernel.Core/Functions/KernelFunctionMetadataFactory.cs:19]: nit: Summary is not correct for what the method is doing.

**Review by alliscode — APPROVED** [review]

**Review by SergeyMenshykh — APPROVED** [review]

File: dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs [hunk 2]
```diff
@@ -100,6 +100,67 @@ public static KernelFunction Create(
...
+            });
+    }
+
+    /// <summary>
+    /// Creates a <see cref="KernelFunctionMetadata"/> instance for a method, specified via an <see cref="MethodInfo"/> instance.
+    /// </summary>
+    /// <param name="method">The method to be represented via the created <see cref="KernelFunction"/>.</param>
+    /// <param name="options">Optional function creation options.</param>
+    /// <returns>The created <see cref="KernelFunction"/> wrapper for <paramref name="method"/>.</returns>
+    public static KernelFunctionMetadata CreateMetadata(
```
Comment (westey-m) [inline dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs:140]: Should this also have the experimental flag?

**Review by westey-m — APPROVED** [review]

**Comment by RogerBarreto** [comment]
I thing this PR or the Issue misses the `Problem / Motivation and Context`

File: dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs [hunk 3]
```diff
@@ -100,6 +100,67 @@ public static KernelFunction Create(
...
+            method,
+            new KernelFunctionFromMethodOptions
+            {
+                FunctionName = functionName,
+                Description = description,
+                Parameters = parameters,
+                ReturnParameter = returnParameter,
+                LoggerFactory = loggerFactory
+            });
+    }
```
Comment (RogerBarreto) [inline dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs:132]:
```suggestion
    => CreateMetadata(
          method,
          new KernelFunctionFromMethodOptions
          {
              FunctionName = functionName,
              Description = description,
              Parameters = parameters,
              ReturnParameter = returnParameter,
              LoggerFactory = loggerFactory
          });
    
```

File: dotnet/src/SemanticKernel.UnitTests/Functions/KernelFunctionMetadataFactoryTests.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.ComponentModel;
+using System.Linq;
+using Microsoft.SemanticKernel;
+using Xunit;
+
+namespace SemanticKernel.UnitTests.Functions;
+public class KernelFunctionMetadataFactoryTests
```
Comment (RogerBarreto) [inline dotnet/src/SemanticKernel.UnitTests/Functions/KernelFunctionMetadataFactoryTests.cs:10]:
```suggestion
namespace SemanticKernel.UnitTests.Functions;

public class KernelFunctionMetadataFactoryTests
```

File: dotnet/src/SemanticKernel.UnitTests/Functions/KernelFunctionMetadataFactoryTests.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    #region private
+#pragma warning disable CA1812 // Used in test case above
+    private sealed class MyKernelFunctions
+    {
+        // Disallow instantiation of this class.
+        private MyKernelFunctions()
+        {
+        }
```
Comment (RogerBarreto) [inline dotnet/src/SemanticKernel.UnitTests/Functions/KernelFunctionMetadataFactoryTests.cs:36]: No constructor needed.

```suggestion
```

**Review by RogerBarreto — APPROVED** [review]

File: dotnet/src/SemanticKernel.UnitTests/Functions/KernelFunctionMetadataFactoryTests.cs (hunk 5 above)
Comment (markwallace-microsoft) [reply dotnet/src/SemanticKernel.UnitTests/Functions/KernelFunctionMetadataFactoryTests.cs:36]: Having a private constructor is part of the test, I want to ensure the code doesn't try to create an instance of the class

**Review by alliscode — APPROVED** [review]

**Pull request merged**